src/Domain/DomainCreators/AlignedBlocks.hpp

```diff
@@ -0,0 +1,93 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <limits>
+#include <vector>
+
+#include "DataStructures/Index.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/DomainCreators/DomainCreator.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/MakeArray.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace DomainCreators {
+
+/// \ingroup DomainCreatorsGroup
+/// \brief Create a Domain consisting of multiple aligned Blocks.
+///
+/// This is useful for setting up problems with piecewise smooth initial data.
+///
+/// \note Adaptive mesh refinement can never join blocks.
+template <size_t VolumeDim, typename TargetFrame>
+class AlignedBlocks : public DomainCreator<VolumeDim, TargetFrame> {
+ public:
+  struct BlockBounds {
+    using type = std::array<std::vector<double>, VolumeDim>;
+    static constexpr OptionString help = {
+        "Coordinates of block boundaries in each dimension."};
+  };
+
+  struct IsPeriodicIn {
+    using type = std::array<bool, VolumeDim>;
+    static constexpr OptionString help = {
+        "Whether the domain is periodic in each dimension."};
+    static type default_value() { return {{false}}; }
```

Comment:
This looks like it doesn't return a default of {{false, false, false}} for 3D

Reply:
Should probably use the `make_array` stuff we have, yes.